code/game/objects/items/storage/belt.dm

```diff
@@ -436,34 +436,21 @@
 		))
 
 /obj/item/storage/belt/grenade/full/PopulateContents()
-	new /obj/item/grenade/flashbang(src)
-	new /obj/item/grenade/smokebomb(src)
-	new /obj/item/grenade/smokebomb(src)
-	new /obj/item/grenade/smokebomb(src)
-	new /obj/item/grenade/smokebomb(src)
-	new /obj/item/grenade/empgrenade(src)
-	new /obj/item/grenade/empgrenade(src)
-	new /obj/item/grenade/syndieminibomb/concussion/frag(src)
-	new /obj/item/grenade/syndieminibomb/concussion/frag(src)
-	new /obj/item/grenade/syndieminibomb/concussion/frag(src)
-	new /obj/item/grenade/syndieminibomb/concussion/frag(src)
-	new /obj/item/grenade/syndieminibomb/concussion/frag(src)
-	new /obj/item/grenade/syndieminibomb/concussion/frag(src)
-	new /obj/item/grenade/syndieminibomb/concussion/frag(src)
-	new /obj/item/grenade/syndieminibomb/concussion/frag(src)
-	new /obj/item/grenade/syndieminibomb/concussion/frag(src)
-	new /obj/item/grenade/syndieminibomb/concussion/frag(src)
-	new /obj/item/grenade/gluon(src)
-	new /obj/item/grenade/gluon(src)
-	new /obj/item/grenade/gluon(src)
-	new /obj/item/grenade/gluon(src)
-	new /obj/item/grenade/chem_grenade/incendiary(src)
-	new /obj/item/grenade/chem_grenade/incendiary(src)
-	new /obj/item/grenade/chem_grenade/facid(src)
-	new /obj/item/grenade/syndieminibomb(src)
-	new /obj/item/grenade/syndieminibomb(src)
-	new /obj/item/screwdriver(src)
-	new /obj/item/multitool(src)
+	var/static/items_inside = list(	
+									/obj/item/grenade/flashbang = 1,
+									/obj/item/grenade/smokebomb = 4,
+									/obj/item/grenade/empgrenade = 1,
+									/obj/item/grenade/empgrenade = 1,
+									/obj/item/grenade/syndieminibomb/concussion/frag = 10,
+									/obj/item/grenade/gluon = 4,
+									/obj/item/grenade/chem_grenade/incendiary = 2,
+									/obj/item/grenade/chem_grenade/facid = 1,
+									/obj/item/grenade/syndieminibomb = 2,
+									/obj/item/screwdriver = 1,
+									/obj/item/multitool = 1)
+	for(var/each_item in items_inside)

```

Comment:
These three lines should probably be in a proc or something instead of being copypasted everywhere.

Reply:
You know, I thought of that, but to be fair I actually dont know how to fix this. If this is a proc, then everything that generates itens should call it. This way we would have a pattern.

Have any code that I could use as an example?

Reply:
```DM
/proc/spawnbyquantity(var/list/your_list)
>>>>for(var/thing in your_list)
>>>>>>>>for(var/timetospawn in 1 to your_list[thing])
>>>>>>>>>>>>new timetospawn
```

Reply:
File organization is one of those subtle things that can be hard to get a grasp on.

Global procs like these can be defined either near their main use case (`proc/PopulateContents` in this case, though that might be tricky since there are two o them), or in one of the files in `code/__HELPERS/`, such as `unsorted.dm` if nothing is quite right.